neo4j/_async/work/session.py

```diff
@@ -543,11 +543,11 @@ async def read_transaction(
             This does not necessarily imply access control, see the session
             configuration option :ref:`default-access-mode-ref`.
 
-        This transaction will automatically be committed unless an exception is thrown during query execution or by the user code.
-        Note, that this function perform retries and that the supplied `transaction_function` might get invoked more than once.
-
-        Managed transactions should not generally be explicitly committed
-        (via ``await tx.commit()``).
+        This transaction will automatically be committed when the function
+        returns, unless an exception is thrown during query execution or by
+        the user code. Note, that this function performs retries and that the
+        supplied `transaction_function` might get invoked more than once.
+        Therefore, it needs to be idempotent (i.e., have no side effects).
```

Comment:
pedantic comment: it _can_ have side effects (like logging e.g.)
it's rather: it should always return the same result no matter how many times it's invoked